docker/install/ubuntu_install_zephyr.sh

```diff
@@ -38,10 +39,12 @@ sudo apt-key add kitware-archive-latest.asc
 
 echo deb https://apt.kitware.com/ubuntu/ bionic main\
      >> /etc/apt/sources.list.d/kitware.list
-sudo apt-get update
+sudo apt-get update --fix-missing
```

Comment:
also it'd be great if we could somehow not apt update if no sources.list.d changed. maybe we could make our script run e.g. `find /etc/apt/sources.list.d /etc/apt/sources.list | xargs md5sum | md5sum` and store the output in /var/apt-update-md5 and only rerun when there is a mismatch?

Reply:
It'd be simpler if we just didn't `apt update` at all and just leave it to happen in the Dockerfile level, would that also work?

Reply:
I am not sure if omitting `apt-update`s is possible. As far as I understand, performing an `apt-get install` will do a lookup in `/var/lib/apt/lists/` for the desired package (and the relevant dependencies). However, after each `apt-get install`, we clear that directory to reduce the docker layer size (resulting in a smaller image). Therefore, I think running `apt-get update` on each install is necessary in this case. 

I tried to implement the md5 sum check and it resulted in [packages not being located](https://ci.tlcpack.ai/blue/organizations/jenkins/tvm/detail/PR-11470/7/pipeline/53).

Also, the [best practices guide for docker files](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run) seems to suggest something similar with regards to running `apt-get update` and `apt-get install` in the same layer.

Please let me know if you have other thoughts.

Reply:
So if I understand correctly, reducing the image size (by removing the files under `/var/lib/apt/lists/`) and decreasing the image build times (and therefore CI times) by not running `apt-get update` each time is not achievable. Also, the recommendation is to run `apt-get update` and clear the cache on every docker layer where apt is used.

Reply:
okay looking over all of this i agree there are some challenges then. I don't think we should take an approach that requires us to run apt-get update for each layer. we do need to run it:
- one time before the first apt-get install for each docker image
- any time a script adds to /etc/apt/sources.list.d

i think we could work around this either by:
- not removing /var/lib/apt/lists (i just built ci_cpu and it's 40M. that seems small relative to any accumulated packages)
- attempting to mount that dir in a docker volume (this doesn't actually look super-feasible from docker build, but perhaps a workaround exists with e.g. buildx or something)
- doing something a bit crazy like making a tar of /var/lib/apt/lists and uploading that to the build host each time. i actually think we should not do this one, but if a simpler similar thing exists and the lists are larger than 40M, maybe it's worth it. 

feel free to investigate a bit here. the main thing i'm interested in is removing any cached debian packages